common/src/lib.rs

```diff
@@ -64,14 +86,95 @@ pub struct Message {
 }
 
 #[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
-pub struct Program {
+pub enum Program {
+    Active(ActiveProgram),
+    Terminated,
```

Comment:
I don't like this naming. The term `Terminated` may confuse about its purpose.

Reply:
Discussed, leave `Terminated`